File: Tower_of_Hanoi.py
print("Starting Tower_of_Hanoi.py")
import time as t
num = int(input("Number of discs n = ")) #select number of discs
def TOH(n,source,dest,aux):
    if n == 1:
        print(f"Move disk 1 from {source} to {dest}.") #shift the nth disk to destination tower from source tower
    else:
        TOH(n-1,source,aux,dest) #shfting all n-1 disks to auxillary tower
        print(f"Move disk {n} from {source} to {dest}.") #shift nth disk to destination tower from source tower
        TOH(n-1,aux,dest,source) #shifting n-1 disks from auxillary tower back to desination treating source tower as new auxillary tower
t_start = t.time()
TOH(num,'A','B','C') #running function with selected number of disks, naming the towers
t_end = t.time()
print(f"Loop ended in {t_end - t_start:.5f} seconds.") #printing time taken for loop to run with a precison of 0.00001s.1
        

File: Tower_of_Hanoi.c
#include <stdio.h>
#include <time.h>

void TOH(int n, char source, char aux, char destination) {
    if (n == 1) {
        printf("Move disk 1 from %c to %c\n", source, destination); // Added newline
    } else {
        TOH(n - 1, source, aux, destination);
        printf("Move disk %d from %c to %c\n", n, source, destination);
        TOH(n - 1, aux, destination, source);
    }
}

int main() {
    int num;

    printf("Enter number of disks to be simulated: ");
    scanf("%d", &num);

    float start_time = clock();

    TOH(num, 'A', 'B', 'C');

    float end_time = clock();

    float elapsed_time = ((double) (end_time - start_time)) / CLOCKS_PER_SEC;
    printf("Time taken to run program for %i disks is %.21f seconds./n", num ,  elapsed_time);

    return 0;
}
